### PR TITLE
【添加】 添加STM32L4R9-EVAL 工程

### PR DESCRIPTION
**Pull request by jhbdream**
Signed-off-by: jhb <jihongbin@rt-thread.com>

## 拉取/合并请求描述：(PR description)

[
这段方括号里的内容是您**必须填写并替换掉**的，否则PR不可能被合并。**方括号外面的内容不需要修改，但请仔细阅读。**
The content in this square bracket must be filled in and replaced, otherwise PR can not be merged. The contents outside square brackets need not be changed, but please read them carefully.

请在这里填写您的PR描述，可以包括以下之一的内容：为什么提交这份PR；解决的问题是什么，你的解决方案是什么；
Please fill in your PR description here, which can include one of the following items: why to submit this PR; what is the problem solved and what is your solution;

并确认并列出已经在什么情况或板卡上进行了测试。
And confirm in which case or board have been tested.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by jhbdream** [comment]
【添加】添加STM32L4R9-EVAL BSP支持

File: bsp/stm32/stm32l4r9-eval/EventRecorderStub.scvd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<component_viewer schemaVersion="0.1" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance" xs:noNamespaceSchemaLocation="Component_Viewer.xsd">
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/EventRecorderStub.scvd]: 这个文件是多余的，请删除。

File: bsp/stm32/stm32l4r9-eval/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# STM32L4R9I-EVAL 开发板bsp说明
+
+## 简介
+
+本文档为 RT-Thread 开发团队为 STM32L4R9I-EVAL 开发板提供的 BSP (板级支持包) 说明。
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/README.md]: 是rtt开发团队做的吗？

File: bsp/stm32/stm32l4r9-eval/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  - 状态指示灯：4个，LED1 - 4
+  - 按键：1个，wakeup（PC13），五向摇杆
+  - TFTLCD 显示屏：2个
+  - 板载 ST LINK
+- 常用接口：SD 卡接口、USB OTG Micro USB 接口、电机接口
+
+- 调试接口，ST-LINK Micro USB 接口
+
+
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/README.md]: 多余行删除。BSP的命名也不对呢，参考其他BSP。

File: bsp/stm32/stm32l4r9-eval/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+#### 硬件连接
+
+使用数据线连接开发板到 PC，打开电源开关。
+
+#### 编译下载
+
+双击 project.uvprojx 文件，打开 MDK5 工程，编译并下载程序到开发板。
+
+> 工程默认配置使用 xxx 仿真器下载程序，在通过 xxx 连接开发板的基础上，点击下载按钮即可下载程序到开发板
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/README.md]: XXX要修改一下。

File: bsp/stm32/stm32l4r9-eval/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+```bash
+ \ | /
+- RT -     Thread Operating System
+ / | \     4.0.1 build Apr 4 2019
+ 2006 - 2019 Copyright by rt-thread team
+msh >
+```
+### 进阶使用
+
+此 BSP 默认只开启了 GPIO 和 串口3的功能，如果需使用 SD 卡、Flash 等更多高级功能，需要利用 ENV 工具对BSP 进行配置，步骤如下：
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/README.md]: SD卡flash改为其他的吧。

File: bsp/stm32/stm32l4r9-eval/template.uvprojx [hunk 6]
```diff
@@ -0,0 +1,395 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
+<Project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="project_projx.xsd">
+
+  <SchemaVersion>2.1</SchemaVersion>
+
+  <Header>### uVision Project, (C) Keil Software</Header>
+
+  <Targets>
+    <Target>
+      <TargetName>rt-thread</TargetName>
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/template.uvprojx]: 名称改为rtthread

File: bsp/stm32/stm32l4r9-eval/template.uvoptx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    <nMigrate>0</nMigrate>
+  </Extensions>
+
+  <DaveTm>
+    <dwLowDateTime>0</dwLowDateTime>
+    <dwHighDateTime>0</dwHighDateTime>
+  </DaveTm>
+
+  <Target>
+    <TargetName>rt-thread</TargetName>
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/template.uvoptx]: 名称改为rtthread

File: bsp/stm32/stm32l4r9-eval/template.ewp [hunk 8]
```diff
@@ -0,0 +1,2031 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project>
+    <fileVersion>3</fileVersion>
+    <configuration>
+        <name>rt-thread</name>
```
Comment (misonyo) [inline bsp/stm32/stm32l4r9-eval/template.ewp]: 名称改为rtthread

**Pull request closed without merging**